axelrod/tournament.py

```diff
@@ -154,14 +138,14 @@ def _run_parallel_repetitions(self, interactions):
         done_queue = Queue()
         workers = self._n_workers()
 
-        for repetition in range(self._parallel_repetitions):
-            work_queue.put(repetition)
+        chunks = self.match_generator.build_match_chunks()
+        for chunk in chunks:
+            # This is crap: I'm going through the generator, ideally want the
```

Comment:
Probably worth removing this comment which was just me when I threw this together quickly as a proof of concept. Given the speedups and memory checks you've done this is perhaps not as bad as I thought at the time.